mlir/include/mlir/Dialect/Ptr/IR/PtrOps.td

```diff
@@ -17,6 +17,75 @@ include "mlir/Interfaces/SideEffectInterfaces.td"
 include "mlir/Interfaces/ViewLikeInterface.td"
 include "mlir/IR/OpAsmInterface.td"
 
+//===----------------------------------------------------------------------===//
+// FromPtrOp
+//===----------------------------------------------------------------------===//
+
+def Ptr_FromPtrOp : Pointer_Op<"from_ptr", [
```

Comment:
I was talking about this with a colleague the other day. We noticed that the memref dialect has `memref.extract_aligned_pointer_as_index`, but it's missing the inverse operation. There's currently no way to construct a memref from a given pointer + metadata. It's good to see some some progress on this.

Reply:
This is by design, the documentation for `memref.extract_aligned_pointer_as_index` states it:

> A reverse operation that creates a memref from an index interpreted as a pointer is explicitly discouraged.

Reply:
The way it's written, it sounds to me like "there's nothing wrong with having an op that creates a memref out of thin air, but until we have a better way than interpreting an index as a pointer, we shouldn't do it".